fiftyone/server/db/_mongo_lightning.py

```
@@ -0,0 +1,629 @@
+    if match_arrays:
+        pipeline += match_arrays
+        if match_search:
+            # match again after unwinding list fields
+            pipeline.append(match_search)
+
+    values = set()
+    exclude = set(query.exclude or [])
+    kwargs = {"hint": query.index} if query.index else {}
+
+    async for value in collection.aggregate(pipeline, **kwargs):
+        value = value.get("_id", None)
```

Ruff (0.15.12)

[warning] `for` loop variable `value` overwritten by assignment target

(PLW2901)